# Generated by Django 3.0.1 on 2019-12-27 20:41

from django.conf import settings
import django.contrib.gis.db.models.fields
from django.db import migrations, models
import django.db.models.deletion
import django.utils.timezone


class Migration(migrations.Migration):

    initial = True

    dependencies = [
        ('mission', '0001_initial'),
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.CreateModel(
            name='GeoImage',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('point', django.contrib.gis.db.models.fields.PointField(geography=True, srid=4326)),
                ('timestamp', models.DateTimeField(default=django.utils.timezone.now)),
                ('deleted', models.BooleanField(default=False)),
                ('deleted_at', models.DateTimeField(blank=True, null=True)),
                ('description', models.TextField()),
                ('original_format', models.CharField(max_length=10)),
                ('priority', models.BooleanField(default=False)),
                ('creator', models.ForeignKey(on_delete=django.db.models.deletion.PROTECT, related_name='creatorimages_geoimage_related', to=settings.AUTH_USER_MODEL)),
                ('deleted_by', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.PROTECT, related_name='deletorimages_geoimage_related', to=settings.AUTH_USER_MODEL)),
                ('mission', models.ForeignKey(null=True, on_delete=django.db.models.deletion.PROTECT, to='mission.Mission')),
                ('replaced_by', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.SET_NULL, to='images.GeoImage')),
            ],
        ),
        migrations.AddIndex(
            model_name='geoimage',
            index=models.Index(fields=['mission', 'priority'], name='images_geoi_mission_520eee_idx'),
        ),
    ]
